acceptance/tests/ensure_puppet-agent_paths.rb

```diff
@@ -0,0 +1,91 @@
+  {:name => :factpath, :posix_expected => '/opt/puppetlabs/puppet/cache/lib/facter:/opt/puppetlabs/puppet/cache/facts', :win_expected => 'C:/ProgramData/PuppetLabs/puppet/cache/lib/facter;C:/ProgramData/PuppetLabs/puppet/cache/facts', :not_path => true},
+  {:name => :module_working_dir, :posix_expected => '/opt/puppetlabs/puppet/cache/puppet-module', :win_expected => 'C:/ProgramData/PuppetLabs/puppet/cache/puppet-module'},
+  {:name => :reportdir, :posix_expected => '/opt/puppetlabs/puppet/cache/reports', :win_expected => 'C:/ProgramData/PuppetLabs/puppet/cache/reports'},
+  {:name => :server_datadir, :posix_expected => '/opt/puppetlabs/puppet/cache/server_data', :win_expected => 'C:/ProgramData/PuppetLabs/puppet/cache/server_data'},
+  {:name => :statedir, :posix_expected => '/opt/puppetlabs/puppet/cache/state', :win_expected => 'C:/ProgramData/PuppetLabs/puppet/cache/state', :installed => :dir},
+  {:name => :yamldir, :posix_expected => '/opt/puppetlabs/puppet/cache/yaml', :win_expected => 'C:/ProgramData/PuppetLabs/puppet/cache/yaml'},
+
+  # logdir/rundir
+  {:name => :logdir, :posix_expected => '/var/log/puppetlabs', :win_expected => 'C:/ProgramData/PuppetLabs/puppet/var/log', :installed => :dir},
+  {:name => :rundir, :posix_expected => '/var/run/puppetlabs', :win_expected => 'C:/ProgramData/PuppetLabs/puppet/var/run', :installed => :dir},
+  {:name => :pidfile, :posix_expected => '/var/run/puppetlabs/agent.pid', :win_expected => 'C:/ProgramData/PuppetLabs/puppet/var/run/agent.pid'},
+]
+
+step 'test configprint outputs'
+agents.each do |agent|
+  config_options.each do |config_option|
+    on(agent, puppet_agent('--configprint ' "#{config_option[:name]}")) do
+      file = if agent['platform'] =~ /win/ then config_option[:win_expected] else config_option[:posix_expected] end
+      assert_match(file, stdout)
+    end
+  end
+end
+
+step 'test puppet genconfig entries'
+agents.each do |agent|
+  on(agent, puppet_agent('--genconfig')) do
+    config_options.each do |config_option|
+      file = if agent['platform'] =~ /win/ then config_option[:win_expected] else config_option[:posix_expected] end
+      assert_match("#{config_option[:name]} = #{file}", stdout)
+    end
+  end
+end
+
+step 'test puppet config paths exist'
+agents.each do |agent|
+  config_options.select {|v| !v[:not_path] }.each do |config_option|
+    path = if agent['platform'] =~ /win/ then config_option[:win_expected] else config_option[:posix_expected] end
+    case config_option[:installed]
+    when :dir
+      if !dir_exists?(agent, path)
+        fail_test("Failed to find expected directory '#{path}' on agent '#{agent}'")
+      end
+    when :file
+      if !file_exists?(agent, path)
+        fail_test("Failed to find expected file '#{path}' on agent '#{agent}'")
+      end
+    end
+  end
+end
+
+step 'test puppet binaries exist'
```

Comment:
yeah.  i was concerned about that.  is there a way we can tell if we're in an AIO or source produced environment?
oh: duh  option[:type]  should tell us if we're aio, git, package, etc